backend/src/services/service-token/service-token-dal.ts

```diff
@@ -28,5 +28,30 @@ export const serviceTokenDALFactory = (db: TDbClient) => {
     }
   };
 
-  return { ...stOrm, findById };
+  const findExpiredTokens = async (tx?: Knex) => {
+    try {
+      const docs: { name: string; projectName: string; createdByEmail: string; id: string; projectId: string }[] =
+        await (tx || db.replicaNode())(TableName.ServiceToken)
+          .leftJoin<TUsers>(
+            TableName.Users,
+            `${TableName.Users}.id`,
+            db.raw(`${TableName.ServiceToken}."createdBy"::uuid`)
+          )
+          .join(TableName.Project, `${TableName.Project}.id`, `${TableName.ServiceToken}.projectId`)
+          .whereRaw(
+            `${TableName.ServiceToken}."expiresAt" < NOW() AND ${TableName.ServiceToken}."notificationSent" = false`
+          )
+          .whereNotNull(`${TableName.Users}.email`)
+          .select(`${TableName.ServiceToken}.name`)
+          .select(`${TableName.ServiceToken}.id`)
+          .select(`${TableName.Project}.name as projectName`)
+          .select(`${TableName.ServiceToken}.projectId`)
+          .select(`${TableName.Users}.email as createdByEmail`);
```

Comment:
nit - please use the `db.ref` pattern here 

<img width="980" alt="image" src="https://github.com/user-attachments/assets/8b623cde-6054-4f49-84e2-a673ba6e0e35" />